adlfs/spec.py

```diff
@@ -118,6 +124,41 @@ def _coalesce_version_id(*args) -> Optional[str]:
         return version_ids.pop()
 
 
+def _create_aio_blob_service_client(
+    account_url: str,
+    location_mode: Optional[str] = None,
+    credential: Optional[str] = None,
+) -> AIOBlobServiceClient:
+    if credential is not None:
+        return AIOBlobServiceClient(
+            account_url=account_url,
+            credential=credential,
+            _location_mode=location_mode,
+            user_agent=_get_user_agent(),
+        )
+    elif location_mode is not None:
+        return AIOBlobServiceClient(
+            account_url=account_url,
+            credential=None,
+            _location_mode=location_mode,
+            user_agent=_get_user_agent(),
+        )
+    else:
+        return AIOBlobServiceClient(
+            account_url=account_url,
+            user_agent=_get_user_agent(),
+        )
```

Comment:
For this factory, it would be interesting to see if we can consolidate the actual client instantiation to one instance. To do this we should be able to leverage keyword arguments, for example:
```python
service_client_kwargs = {
    "account_url": account_url,
    "user_agent": _get_user_agent(),
}
if credential is not None:
    service_client_kwargs["credentials"] = credentials
if location_mode is not None:
    service_client_kwargs["_location_mode"] = location_mode
return AIOBlobServiceClient(**service_client_kwargs)
```
The main reason that I'm suggesting we do this is it reduces the verbosity and makes it easier to follow/add to the logic when there is only one path to instantiating the client.